skattoju/openshift-preflight
# Notify users if there is a new version of preflight

- chore: fix typo in log message
- adding platform option to generateContainerCheckOptions
- Bump github.com/spf13/viper from 1.14.0 to 1.15.0
- Bump github.com/onsi/gomega from 1.24.2 to 1.26.0
- Bump k8s.io/client-go from 0.26.0 to 0.26.1
- configure the insecure flag to allow access to registries with self-signed certificate
- fix deployable_by_olm to prioritize installmodes
- Bump github.com/google/go-containerregistry from 0.12.1 to 0.13.0
- Bump github.com/onsi/ginkgo/v2 from 2.7.0 to 2.8.0
- Bump sigs.k8s.io/controller-runtime from 0.14.1 to 0.14.4
- Bump github.com/docker/cli
- removing oc as a binary dependency since we now use the library
- Bump github.com/spf13/afero from 1.9.3 to 1.9.4
- adding CLoseWithError to crane.Export for cases when reader returns EOF, also changing the untar to a ReadCloser so we can explicitly close
- Bump github.com/onsi/gomega from 1.26.0 to 1.27.1
- updating openshift api and go-client dependencies
- updating the readme to point to offical certification docs, and cleaning up the testing section of the document
- Update go-rpmdb
- Bump github.com/onsi/gomega from 1.27.1 to 1.27.2
- updating scorecard knowledge base urls to point to new document location
- Bump sigs.k8s.io/controller-runtime from 0.14.4 to 0.14.5
- Bump k8s.io/client-go from 0.26.1 to 0.26.2
- change dependabot to weekly checks to reduce the noise
- Bump github.com/onsi/ginkgo/v2 from 2.8.4 to 2.9.0
- Bump github.com/spf13/afero from 1.9.4 to 1.9.5
- adding DeferCleanup to check container tests to reset certification project id
- refactoring check container rune to use a closure which now accepts a func with the same signature as cli.RunPreflight so the check is more testable
- Bump github.com/onsi/gomega from 1.27.2 to 1.27.4
- Switch go-rpmdb to use the upstream
- incorporating runPreflight func call into check_operator cmd to allow for better testability
- Bump k8s.io/client-go from 0.26.2 to 0.26.3
- Bump github.com/google/go-containerregistry from 0.13.0 to 0.14.0
- ensure the engine logs error strings
- use a local viper instance for the CLI
- Add experimental support for helm chart certification
- Bump github.com/onsi/gomega from 1.27.4 to 1.27.5
- Bump github.com/glebarez/go-sqlite from 1.20.3 to 1.21.0
- Sign release images with keyless cosign
- setting up log formatter early to force a proper format even if os.OpenFile fails
- remove verbosity from target image log line so the image registry/repo are always included in logs
- Use mutate.Extract for all images
- Fix HasModifiedFiles to check layer by layer
- implementing `--offline` flag, to tar up the artifacts directory
- updating operator-sdk to 1.28.0
- Bump sigs.k8s.io/controller-runtime from 0.14.5 to 0.14.6
- Bump github.com/onsi/gomega from 1.27.5 to 1.27.6
- Bump github.com/docker/cli
- Bump github.com/glebarez/go-sqlite from 1.21.0 to 1.21.1
- Bump github.com/go-logr/logr from 1.2.3 to 1.2.4
- Remove lib from actions and workflows
- Bump github.com/spf13/cobra from 1.6.1 to 1.7.0
- Bump github.com/docker/cli
- Fix tests writing to filesystem
- Test that config file is read from
- removing dependency on ocp-olm-catalog-validator, since calling operator-sdk AlphaDeprecatedAPIsValidator gives us the same validation
- Process links in layers to ignore the target
- Ignore layer files that have security capability attrs
- Bump operator-sdk to 1.28.1 to unlock the lastest Preflight for DCI
- Revert "Bump operator-sdk to 1.28.1 to unlock the lastest Preflight for DCI"
- Do not fail based on reinstall
- Update golangci-lint to 1.52.2
- Resolve race condition in test execution preventing multiple tests from asserting config state
- Update experimental chart-verifier version to align with their 1.11.0 release
- Bump github.com/operator-framework/api
- Ignore build-id directory in HasModifiedFiles
- always send registry/repo info the pyxis for validation
- Add ability to ignore HasModifiedFiles for a given prefix and suffix
- add scratch policy exception support for os_content_type
- Bump github.com/sirupsen/logrus from 1.9.0 to 1.9.3
- Update setup-go in Actions to v4
- Replace 'deadcode' with 'unused' in golangci-lint
- Update coveralls in CI to v2
- Bump github.com/google/go-containerregistry from 0.14.0 to 0.15.2
- Exclude usr/tmp from HasModifiedFiles
- Move to Operator-SDK 1.29.0 containing fix for disconnected env
- Bump github.com/docker/cli
- Bump github.com/onsi/ginkgo/v2 from 2.9.2 to 2.10.0
- updating k8s dependencies and go version, since dependabot can't make all these changes at once
- Bump github.com/onsi/gomega from 1.27.7 to 1.27.8
- conditionally calling pyxis.WithRPMManifest based on the policy that's available in the context and excluding scatch policy from pyxis.WithRPMManifest
- Bump github.com/operator-framework/api from 0.17.5 to 0.17.6
- Bump github.com/operator-framework/operator-manifest-tools
- Bump github.com/spf13/viper from 1.15.0 to 1.16.0
- updating to ubi 9
- update cert_document URL
- Make .cache exclusion broader
- removing unused bundle validate code that relied on operator-sdk binary
- move latest ocp version to 4.13
- explicitly set CGO_ENABLED=0 in the build command
- Updated bundle fails scorecard-test to v1.30.0
- check the action for published state and for the existence of the release object
- Bump github.com/onsi/ginkgo/v2 from 2.10.0 to 2.11.0
- Check policy directly in submit
- Use single quotes for GHA expressions
- Move to Operator-SDK 1.30.0 based on ubi8.8
- Ignore ghost and artifact RPM files
- Add 'github-actions' config to dependabot
- adding rpm architecture of to our maps, so 32 bit and 64 bit rpms do not fail has modified files check
- Bump actions/setup-go from 3 to 4
- Bump gotest.tools/v3 from 3.4.0 to 3.5.0
- Bump shrink/actions-docker-extract from 1 to 2
- Bump github.com/docker/cli
- Bump github.com/onsi/gomega from 1.27.8 to 1.27.10
- Bump sigstore/cosign-installer
- Bump github.com/glebarez/go-sqlite from 1.21.1 to 1.21.2
- Refactor options code to its own package
- Refactor CraneEngine to craneEngine
- Process image manifest if platform not specified
- Enable groups for dependabot
- Bump github.com/operator-framework/api from 0.17.6 to 0.17.7
- Bump sigstore/cosign-installer
- Bump github.com/operator-framework/operator-manifest-tools
- Move to Operator-SDK 1.31.0 using Ansible 2.15.0
- updating ginko dependency
- Bump actions/checkout from 3 to 4
- Bump shrink/actions-docker-extract from 2 to 3
- Bump sigstore/cosign-installer
- Bump docker/login-action from 2 to 3
- Bump docker/setup-qemu-action from 2 to 3
- Bump sigstore/cosign-installer
- updating some dependencies
- feat: add --config flag to allow different config files to be passed
- Updating the document URL
- Updating doc URL
- updating doc URL to the latest
- doc URL update to 8.69
- updating to operator-sdk v1.32.0
- Do not process attestation manifests
- updating ocp to k8s map for latest release
- updating gomega to version 1.28.0
- updating go to 1.21
- Bump AButler/upload-release-assets from 2.0 to 3.0
- add check for latest available version
